onnx2torch/node_converters/pad.py

```diff
@@ -0,0 +1,126 @@
+__all__ = [
+    'OnnxPadStatic',
+    'OnnxPadDynamic',
+]
+
+from typing import List
+from typing import Optional
+
+import torch
+import torch._C as torch_C
+from torch import nn
+
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+from onnx2torch.utils.common import OnnxMapping
+from onnx2torch.utils.common import OnnxToTorchModule
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx
+from onnx2torch.utils.custom_export_to_onnx import OnnxToTorchModuleWithCustomExport
+
+
+class OnnxPadStatic(nn.Module, OnnxToTorchModule):
+
+    def __init__(
+        self,
+        pads: List[int],
+        mode: str = 'constant',
+        constant_value: float = 0.0,
+    ):
+        super().__init__()
+        self.mode = mode
+        self.pads = pads
+        self.constant_value = constant_value
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        return torch.nn.functional.pad(
+            input_tensor,
+            mode=self.mode,
+            pad=self.pads,
+            value=self.constant_value,
+        )
+
+
+class OnnxPadDynamic(nn.Module, OnnxToTorchModuleWithCustomExport):
+
+    def __init__(self, mode: str = 'constant'):
+        super().__init__()
+        self.mode = mode
+
+    @staticmethod
+    def _do_forward(
+        input_tensor: torch.Tensor,
+        mode: str,
+        pads: torch.Tensor,
+        constant_value: Optional[float],
+    ) -> torch.Tensor:
+
+        if constant_value is not None:
+            return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads, value=constant_value)
+
+        return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads)
+
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        pads: torch.Tensor,
+        constant_value: Optional[float] = None,
+    ) -> torch.Tensor:
+
+        if torch.unique(pads).shape[0] > 1:
```

Comment:
If we want to support different paddings we need to change `pads` formats.
onnx format: [x1_begin, x2_begin, ..., x1_end, x2_end, ...] -> torch format [x_n-1_begin, x_n-1_end, x_n-2_begin, x_n-2_end, ...]
```python
# Let ends is defined
middle = len(pads) / 2
# middle should be integer otherwise raise error
onnx_pad_begin, onnx_pad_end = pads[:middle], pads[middle:]
# invert dimensions
onnx_pad_begin, onnx_pad_end = onnx_pad_begin[::-1], onnx_pad_end[::-1]
torch_pads = []
for b, e in zip(onnx_pad_begin, onnx_pad_end):
    torch_pads.extend([b, e])
``` 
You can reduce/simplify this code.

Reply:
Now we don't support different paddings

Reply:
Done